docs/examples/DataSet/Comparing qcodes Measurement context manager with qcodes doNd functions.ipynb

```diff
@@ -0,0 +1,443 @@
+    "from qcodes.utils.dataset import doNd\n",
+    "from qcodes.dataset.plotting import plot_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a database or loading it"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is not required to have a database for this notebook, but we would like to have it. We encourage users to open a plottr and view the results of this notebook using its live plotting feature (more info [How to use plottr with QCoDeS for live plotting](../plotting/How-to-use-Plottr-with-QCoDeS-for-live-plotting.ipynb))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Upgrading database; v0 -> v1: : 0it [00:00, ?it/s]\n",
+      "Upgrading database; v1 -> v2: 100%|█████████████████████████████████████████████████████| 1/1 [00:00<00:00, 250.20it/s]\n",
+      "Upgrading database; v2 -> v3: : 0it [00:00, ?it/s]\n",
+      "Upgrading database; v3 -> v4: : 0it [00:00, ?it/s]\n",
+      "Upgrading database; v4 -> v5: 100%|█████████████████████████████████████████████████████| 1/1 [00:00<00:00, 251.59it/s]\n",
+      "Upgrading database; v5 -> v6: : 0it [00:00, ?it/s]\n",
+      "Upgrading database; v6 -> v7: 100%|█████████████████████████████████████████████████████| 1/1 [00:00<00:00, 125.00it/s]\n",
+      "Upgrading database; v7 -> v8: 100%|█████████████████████████████████████████████████████| 1/1 [00:00<00:00, 250.35it/s]\n",
+      "Upgrading database; v8 -> v9: 100%|█████████████████████████████████████████████████████| 1/1 [00:00<00:00, 249.81it/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "initialise_or_create_database_at('doNd.db')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### One-dimensional measurement"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's start with a simple one-dimensional case, where we sweep over the parameter 'x' and retrieve parameter 'y' and 'z' calculated as cos and sin on those swept 'x' points, repectively. We intentionally want to have two independent 'y' and 'z' calculated parameters to show how we can have more than one output parameter in the Measurement and doNd."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Measurement"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Starting experimental run with id: 1. \n",
+      "Meaurement took 5.572435140609741 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Setting up parameters\n",
+    "x = Parameter('x', set_cmd=None)\n",
+    "y = Parameter('y', get_cmd=None)\n",
+    "z = Parameter('z', get_cmd=None)\n",
```

Comment:
To make this clearer we should use the same parameters in all examples. E.g. add the lambda functions here and reuse these parameters throughout the examples

Reply:
This is a good idea. But, there should be two sets of parameters, one for 1D and the other one for 2D because parameters are slightly different in two section (only x is the same in all cases). I can segregate them by saying x1, y1, z1 and x2, y2 and z2.  What do you think?

Reply:
I suggest that you create all the parameters up front such that you have something like the following:
Otherwise y will change from a independent to a dependent parameter.
```
x = Parameter('x', set_cmd=None)
y = Parameter('y', set_cmd=None)
z1 = Parameter('z1', get_cmd=lambda: np.sin(x())) 
z3 = Parameter('z2', get_cmd=lambda: np.cos(x())
z3 = Parameter('z1', get_cmd=lambda: np.cos(x())*np.sin(y())) 
```

Reply:
This is much cleaner, thanks.

Reply:
@FarBo if this has been done please feel free to press the resolve conversation button ;) That makes it easier to see if this is ready for a new review